### ISSUE TITLE
Any plans to support Apache arrow bindings?

### ISSUE THREAD

**Issue by chitralverma** [cloudera/impyla#507]

**Comment by csringhofer** [comment]
You mean extending the cursor to return fetch results in Arrow format instead of the current row oriented way?

I don't know of any plans but it sounds like a good addition to Impyla.

**Comment by chitralverma** [comment]
> You mean extending the cursor to return fetch results in Arrow format instead of the current row oriented way?
> 
> I don't know of any plans but it sounds like a good addition to Impyla.

Yes, it will be great if we could have `as_pyarrow_table` and `as_pyarrow_dataset` options available some where to return the results as a PyArrow Table (eagerly) or PyArrow Dataset (lazyily) which doing zero-copy.

**Comment by Khalid-Nowaf** [comment]
I would second this strongly. +1 

I'm not a Python guy, but I'm using this since it is the only (client/driver) I know for Impala that is stable and feature-complete. Adding arrow data format support will allow us to wrap it in different languages/systems with minimal cost.
